src/System.Net.Security/src/System/Net/CertificateValidationPal.OSX.cs

```diff
@@ -0,0 +1,145 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Net
+{
+    internal static partial class CertificateValidationPal
+    {
+        internal static SslPolicyErrors VerifyCertificateProperties(
+            SafeDeleteContext securityContext,
+            X509Chain chain,
+            X509Certificate2 remoteCertificate,
+            bool checkCertName,
+            bool isServer,
+            string hostName)
+        {
+            SslPolicyErrors errors = SslPolicyErrors.None;
+
+            if (remoteCertificate == null)
+            {
+                errors |= SslPolicyErrors.RemoteCertificateNotAvailable;
+            }
+            else
+            {
+                if (!chain.Build(remoteCertificate))
+                {
+                    errors |= SslPolicyErrors.RemoteCertificateChainErrors;
+                }
+
+                if (!isServer && checkCertName)
+                {
+                    SafeDeleteSslContext sslContext = (SafeDeleteSslContext)securityContext;
+
+                    if (!Interop.AppleCrypto.SslCheckHostnameMatch(sslContext.SslContext, hostName))
+                    {
+                        errors |= SslPolicyErrors.RemoteCertificateNameMismatch;
+                    }
+                }
+            }
+
+            return errors;
+        }
+
+        //
+        // Extracts a remote certificate upon request.
+        //
+        internal static X509Certificate2 GetRemoteCertificate(
```

Comment:
(maybe it was github compare collapsing some diffs - didn't see it - and hate that feature)